### PR TITLE
fixed spatial index bug

### PR DESCRIPTION
**Pull request by mmicek**

### CODE REVIEW

**Comment by mmicek** [comment]
Fix for bug: https://github.com/kaaengine/kaa/issues/132

File: demos/CMakeLists.txt [hunk 1]
```diff
@@ -35,3 +35,4 @@ add_demo(demo-window window.cpp)
 add_demo(demo-drawing-order drawing_order.cpp)
 add_demo(demo-spatial-indexing spatial_indexing.cpp)
 add_demo(demo-input input.cpp)
+add_demo(demo-index-spatial index_spatial.cpp)
```
Comment (maniek2332) [inline demos/CMakeLists.txt]: rename to: `demo-spatial-indexing-2` and `spatial_indexing_2` (or similiar)

File: demos/index_spatial.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+#include <glm/glm.hpp>
+
+#include "kaacore/engine.h"
+#include "kaacore/log.h"
+#include "kaacore/nodes.h"
+#include "kaacore/scenes.h"
+
+struct TestDemoScene : kaacore::Scene {
+    std::vector<glm::dvec2> points;
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused variable

File: demos/index_spatial.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    kaacore::NodePtr shape_repr;
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused variable

File: demos/index_spatial.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        auto shape_repr = kaacore::make_node();
+        auto shape_repr2 = kaacore::make_node();
+
+        shape_repr2->position(glm::dvec2(75, 75));
+        shape_repr2->shape(kaacore::Shape::Box({150, 150}));
+
+        shape_repr->position(glm::dvec2(0, 0));
+        shape_repr->shape(kaacore::Shape::Box({150, 150}));
+        shape_repr->add_child(shape_repr2);
+        this->shape_repr = this->root_node.add_child(shape_repr);
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused variable

File: demos/index_spatial.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    void update(const kaacore::Duration dt) override
+    {
+        for (auto const& event : this->get_events()) {
+            if (auto mouse_button = event.mouse_button()) {
+                auto pos = mouse_button->position();
+                if (mouse_button->button() == kaacore::MouseButton::left) {
+                    pos = this->camera().unproject_position(pos);
+                    auto query = this->spatial_index.query_point(pos);
+                    KAACORE_LOG_DEBUG("Number of nodes: {}", query.size());
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: use `KAACORE_APP_LOG_INFO` to log application related messages

File: demos/index_spatial.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
+#include <cstdlib>
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused include

File: demos/index_spatial.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+#include <vector>
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused include

File: demos/index_spatial.cpp [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+#include <glm/glm.hpp>
```
Comment (maniek2332) [inline demos/index_spatial.cpp]: unused include

**Review by maniek2332 — CHANGES_REQUESTED** [review]

**Review by maniek2332 — APPROVED** [review]

**Pull request merged**